Added code to process eTRV 'TEMPERATURE' OpenThings monitor message

genestealer opened the pull request:
Details: xxx

Fixes
https://github.com/Achronite/mqtt-energenie-ener314rt/issues/96

How it looks:
![image](https://github.com/user-attachments/assets/2a353170-3735-4007-941c-8f0891897e47)
![image](https://github.com/user-attachments/assets/ca76cefa-5cff-4691-a366-3874d7cee2a2)
![image](https://github.com/user-attachments/assets/16c2aa89-b6d4-4b16-8c63-44f9371e464c)
![image](https://github.com/user-attachments/assets/0db057fe-bf54-448e-b418-caa092f3ff70)

genestealer commented:
Also added Add dropdown to climate control dashboard card for most used maintainance operations for eTRV
https://github.com/Achronite/mqtt-energenie-ener314rt/issues/90
![image](https://github.com/user-attachments/assets/b71235b9-4cf2-45d7-989b-2acd85672783)

Achronite commented on README.md:
```diff
@@ -325,7 +325,7 @@ Both MiHome heating devices are now supported (as of v0.7.x).  Specifically the
 
 ### Command Caching
-Battery powered energenie devices, such as the eTRV or Thermostat do not constantly listen for commands.  For example, the eTRV reports its temperature at the *SET_REPORTING_INTERVAL* (default 5 minutes) after which the receiver is then activated to listen for commands. The receiver only remains active for 200ms or until a message is received.
+Battery powered energenie devices, such as the eTRV or Thermostat do not constantly listen for commands.  For example, the eTRV reports its temperature at the *SET_REPORT_PERIOD* (default 5 minutes) after which the receiver is then activated to listen for commands. The receiver only remains active for 200ms or until a message is received.
```
Change to REPORT_PERIOD (remove SET)

Achronite commented:
Sorry for the delay, with Christmas and all that!

I'm not entirely sure about the use of cached temperature and setting of state to show 'heating' or not; surely this is also reliant upon if the boiler has the house heating on or not?

I must apologise though; I modified my code to add 'auto' on my climate nodes quite a while back, after installing the Drayton Wiser system and observing what it did; I just hadn't pushed them to 'develop' yet.  

3.json:
```
      {
         "id": "Climate_Control",
         "component": "climate",
         "main": true,
         "config": {
            "curr_temp_t": "~TEMPERATURE/state",
            "max_temp": 40,
            "min_temp": 5,
            "modes": [
               "auto",
               "heat",
               "off"
            ],
            "mode_cmd_t": "~VALVE_STATE/command",
            "mode_stat_t": "~VALVE_STATE/state",
            "mode_cmd_tpl": "{% if value == 'off' %} 1 {% elif value == 'auto' %} 2 {% else %} 0 {% endif %}",
            "mode_stat_tpl": "{% if value == '1' %} off {% elif value == '2' %} auto {% else %} heat {% endif %}",
            "temp_cmd_t": "~TARGET_TEMP/command",
            "temp_stat_t": "~TARGET_TEMP/state",
            "ic": "mdi:radiator",
            "opt": "true",
            "temp_step": "0.5",
            "act_tpl": "{% if value == 'Off' %} off {% else %} heating {% endif %}",
            "act_t": "heating/state"
         }
```

I also added an automation to set the 'heating' status for auto mode based on the boiler state using MQTT from within HA:
![image](https://github.com/user-attachments/assets/7bfc97f6-2799-4f29-a699-c00e8bd0ae31)

I'll push the changes I made to develop now.

genestealer commented:
Hi @Achronite,

> I'm not entirely sure about the use of cached temperature and setting of state to show 'heating' or not; surely this is also reliant upon if the boiler has the house heating on or not?


The valve action state (heating/idle) for each valve is designed to be independent of external factors, including other valves and the heat source/boiler. This approach aligns with how other MQTT radiator valve integrations work. The retained temperature set-point in MQTT is used as it is reliably set by the app.js code when the valve is commanded to a new target temperature. This ensures consistent state determination for each valve.

> I must apologise though; I modified my code to add 'auto' on my climate nodes quite a while back, after installing the Drayton Wiser system and observing what it did; I just hadn't pushed them to 'develop' yet.


The "auto" mode is not suitable for the Energenie eTRVs because these valves do not have a stored schedule. According to Home Assistant’s climate documentation, "Auto: The device is set to a schedule, learned behavior, AI." I removed all options except "heat" to simplify functionality and avoid confusion.

Including "off" can also be misleading, as it suggests "valve closed," but without command caching, this could cause issues. For example:

- If the mode is set to "heat" and then a target temperature is changed, the new target temperature will override the mode. This could result in the valve staying closed (off) while appearing to be set to "heat," leading to inconsistent behavior.

> I also added an automation to set the 'heating' status for auto mode based on the boiler state using MQTT from within HA:


If I understand correctly, this means setting all valve states to heating whenever the boiler is on, regardless of whether individual valves are "calling for heat" (i.e., the set-point temperature is higher than the current room temperature). This is an interesting approach and could be useful to ensure coordination between the boiler and valves. However, it diverges from the idea of each valve independently reporting its own status. Allowing valves to reflect their individual heating or idle state independently might offer more granular control, while additional logic for boiler coordination could live in Home Assistant automations.

genestealer commented:
@Achronite 
Please note, according to Home Assistant’s climate documentation, the HVAC action (`act_tpl`) you have used is 'heating' & 'off' but I think you should have used 'idle' rather than 'off' see [documentation](https://developers.home-assistant.io/docs/core/entity/climate/#hvac-action). 

> The HVAC action describes the current action. This is different from the mode, because if a device is set to heat, and the target temperature is already achieved, the device will not be actively heating anymore.

Achronite commented:
> @Achronite 
> Please note, according to Home Assistant’s climate documentation, the HVAC action (`act_tpl`) you have used is 'heating' & 'off' but I think you should have used 'idle' rather than 'off' see [documentation](https://developers.home-assistant.io/docs/core/entity/climate/#hvac-action). 
> 
> > The HVAC action describes the current action. This is different from the mode, because if a device is set to heat, and the target temperature is already achieved, the device will not be actively heating anymore.
> 
> 
> 

Ok.

genestealer commented:
Hello @Achronite 

This is now tested and ready to go 😊

genestealer commented:
@Achronite Any chance you could have a look at merging this PR please?

genestealer commented on devices/3.json:
```diff
@@ -11,21 +11,72 @@
-            "mode_cmd_tpl": "{% if value == 'off' %} 1 {% elif value == 'auto' %} 2 {% else %} 0 {% endif %}",
-            "mode_stat_tpl": "{% if value == '1' %} off {% elif value == '2' %} auto {% else %} heat {% endif %}",
+            "mode_cmd_tpl": "{% if value == 'off' %} 1 {% else %} 2 {% endif %}",
+            "mode_stat_tpl": "{% if value == '1' %} off {% else %} heat {% endif %}",
```
The nomenclature of 'auto' is typically used for TRV's which have a built-in schedule. Heat / Off is the correct approach for this simple TRV's. 'Normal' is also what is called up in the underlaying energenie-ener314rt node.js module https://github.com/Achronite/energenie-ener314rt/blob/681b5bc1c8b2b2830a6f48bcb9560180e9367989/C/achronite/openThings.h#L51 for the mode that is not "fully open" or "fully closed"

genestealer commented on config_sample.json:
```diff
@@ -2,15 +2,18 @@
-    "ook_xmits": 10,
-    "fsk_xmits": 5,
+    "ook_xmits": 20,
+    "fsk_xmits": 20,
```
This matches the default values

genestealer commented on app.js:
```diff
@@ -676,9 +1317,48 @@ forked.on("message", msg => {
+						}
+						// Set retain flag for TARGET_TEMP
+						retain = true;
+						break;
```
Fixed

genestealer commented on README.md:
```diff
@@ -369,10 +415,10 @@ To support the MiHome Radiator Valve (MIHO013) aka **'eTRV'**, additional code h
-|REPORTING_INTERVAL|Frequency the eTRV will work up and report (in seconds)|command|300-3600|Number|
+|REPORT_PERIOD|Frequency the eTRV will work up and report (in seconds)|command|300-3600|Number|
 |TARGET_TEMP|Target temperature in celcius|state,command|5.0 to 40.0<br>0.5 increments|Number|
 |TEMPERATURE|The current temperature in celcius|state|float|sensor|
```
Fixed
